Gems/Atom/RPI/Code/Source/RPI.Public/Pass/Specific/DownsampleSinglePassMipChainPass.cpp

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Atom/RHI/Factory.h>
+#include <Atom/RPI.Public/Pass/Specific/DownsampleSinglePassMipChainPass.h>
+#include <Atom/RPI.Reflect/Buffer/BufferAssetCreator.h>
+
+// For "A_CPU" macro, refer ffx_a.h and below:
+// https://github.com/GPUOpen-Effects/FidelityFX-SPD/blob/c52944f547884774a1b33066f740e6bf89f927f5/sample/src/DX12/SPDIntegration.hlsl
+#define A_CPU
+#ifdef _MSC_VER
+#pragma warning(disable: 4505) // unreferenced local function has been removed.
+#endif
+#include <ffx_a.h>
+#include <ffx_spd.h>
+
+namespace AZ
+{
+    namespace RPI
+    {
+        Ptr<DownsampleSinglePassMipChainPass> DownsampleSinglePassMipChainPass::Create(const PassDescriptor& descriptor)
+        {
+            Ptr<DownsampleSinglePassMipChainPass> pass = aznew DownsampleSinglePassMipChainPass(descriptor);
+            return pass;
+        }
+
+        DownsampleSinglePassMipChainPass::DownsampleSinglePassMipChainPass(const PassDescriptor& descriptor)
+            : ComputePass(descriptor)
+        {
+            BuildGlobalAtomicBuffer();
+        }
+
+        void DownsampleSinglePassMipChainPass::BuildInternal()
+        {
+            GetInputInfo();
+            CalculateBaseSpdImageSize();
+            BuildPassAttachment();
+            ComputePass::BuildInternal();
+        }
+
+        void DownsampleSinglePassMipChainPass::ResetInternal()
+        {
+            m_indicesAreInitialized = false;
+            m_mipsIndex.Reset();
+            m_numWorkGroupsIndex.Reset();
+            m_workGroupOffsetIndex.Reset();
+            m_imageSizeIndex.Reset();
+            m_inputOutputImageIndex.Reset();
+            m_mip6ImageIndex.Reset();
+            m_globalAtomicIndex.Reset();
+            ComputePass::ResetInternal();
+        }
+
+        void DownsampleSinglePassMipChainPass::FrameBeginInternal(FramePrepareParams params)
+        {
+            SetConstants();
+
+            if (m_globalAtomicBuffer)
+            {
+                // Clear Global Atomic.
+                auto* buffer = static_cast<SpdGlobalAtomicBuffer*>(m_globalAtomicBuffer->Map(sizeof(SpdGlobalAtomicBuffer), 0));
```

Comment:
If I remember correctly, this only needs to be cleared on the first frame correct?

Reply:
You are right.  I removed this map/unmap and make it clear by CommonBufferDescriptor::m_bufferData.